pkg/plugins/common/kustomize/v2-alpha/scaffolds/init.go

```diff
@@ -62,23 +62,45 @@ func (s *initScaffolder) Scaffold() error {
 		machinery.WithConfig(s.config),
 	)
 
-	return scaffold.Execute(
-		&rbac.Kustomization{},
-		&rbac.AuthProxyRole{},
-		&rbac.AuthProxyRoleBinding{},
-		&rbac.AuthProxyService{},
-		&rbac.AuthProxyClientRole{},
-		&rbac.RoleBinding{},
-		&rbac.LeaderElectionRole{},
-		&rbac.LeaderElectionRoleBinding{},
-		&rbac.ServiceAccount{},
-		&manager.Kustomization{},
-		&manager.Config{Image: imageName},
-		&manager.ControllerManagerConfig{},
-		&kdefault.Kustomization{},
-		&kdefault.ManagerAuthProxyPatch{},
-		&kdefault.ManagerConfigPatch{},
-		&prometheus.Kustomization{},
-		&prometheus.Monitor{},
-	)
+	if s.config.IsComponentConfig() {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&manager.ControllerManagerConfig{},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+	} else {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
```

Comment:
```suggestion
			&kdefault.ManagerAuthProxyPatch{},
			&prometheus.Kustomization{},
```

Reply:
Even, if we keep these file as it is, it will not have any problem.

Reply:
Started working on it.